pkg/reconciler/revision/revision.go

```diff
@@ -78,18 +78,31 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		ServiceAccountName: rev.Spec.ServiceAccountName,
 		ImagePullSecrets:   imagePullSecrets,
 	}
-	digest, err := c.resolver.Resolve(rev.Spec.GetContainer().Image,
-		opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-	if err != nil {
-		err = fmt.Errorf("failed to resolve image to digest: %w", err)
-		rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-			v1.RevisionContainerMissingMessage(
-				rev.Spec.GetContainer().Image, err.Error()))
-		return err
+	if rev.Status.ImageDigests == nil {
+		rev.Status.ImageDigests = make(map[string]string, len(rev.Spec.Containers))
 	}
+	for _, container := range rev.Spec.Containers {
+		digest, err := c.resolver.Resolve(container.Image,
+			opt, cfgs.Deployment.RegistriesSkippingTagResolving)
+		if err != nil {
+			err = fmt.Errorf("failed to resolve image to digest: %w", err)
+			rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
+				v1.RevisionContainerMissingMessage(
+					container.Image, err.Error()))
+			return err
+		}
+		if len(rev.Spec.Containers) == 1 || len(container.Ports) != 0 {
+			rev.Status.ImageDigest = digest
+		}
 
-	rev.Status.ImageDigest = digest
-
+		// In order to keep consistency with original behavior of single container handled below scenario
+		// revision status should not contains imageDigests field in these scenario
+		// 1. If flag is disabled
+		// 2. If flag enabled but applied revision has single container
+		if cfgs.Defaults.EnableMultiContainer && len(rev.Spec.Containers) > 1 {
```

Comment:
Why wouldn't we switch to the map basically always? Even if the MC is disabled. The map will just have a single entry.

Reply:
Yes if MC is disabled map will have just one entry.

But as per the current behavior the revision status don't have **ImageDigests** field 
In that case if we don't add that particular condition even if MC disabled revision will have **ImageDigests** field .

So in order to keep consistency with current behavior of revision status added this logic

Let me know allowing **ImageDigests**  field even if MC disabled is okay then i will update code

Reply:
Yeah, I think it is totally fine to have this map populated as source of truth always going forward.

Reply:
I agree, let's add it always.